src/aleph/http/client.clj

```diff
@@ -609,6 +614,16 @@
        ([_ ctx]
          (let [ch (.channel ctx)]
            (reset! in (netty/buffered-source ch (constantly 1) 16))
+
+           ;; Start handshake timeout timer
+           (reset! timeout-task
+                   (.in time/*clock*
```

Comment:
One another thing... Maybe it's better to use EventLoop executor to schedule the task (`ctx.executor().schedule`) instead of manifold's clock? We can also just trigger user event from the runnable and check if the handshake was complete in `:user-triggered`. It seems like this approach would leave fewer spaces for race conditions. What do you think? @shilder  @ztellman 

I still wonder why this functionality is not implemented in Netty, e.g. `SslHandler` supports handshake timeout as a parameter. I'm going to open an issue to start a discussion there.

Reply:
>One another thing... Maybe it's better to use EventLoop executor to schedule the task (ctx.executor().schedule) instead of manifold's clock

Yes, we can use `EventExecutor` and Netty is using it internally in `IdleStateHandler` to schedule timeout handlers.

And about `:user-triggered` - yes, it will work, but I'm not sure how it will help with race conditions ? AFAIU `Future#cancel` provides synchronization point. Or maybe you are talking about situation where context is closed by timeout handler while another thread awaits on `Future#cancel` ? I'm not a Netty expert, so I'm not sure how it handles serialization of events for context

Reply:
Not only for IdleTimeoutHandler but for all timeout tasks. Future canceling is what I want to avoid here specifically because of a synchronization: it might hurt performance when you introduce sync primitives to the pipeline running on I/O thread. 

If you don’t mind, I can update your PR.

Reply:
On it!